build.go

```diff
@@ -97,6 +102,16 @@ func (vs *version) addFiles(partitions map[int]bool) error {
 }
 
 func (vs *version) addFile(file string, partitions map[int]bool) error {
+	if vs.stats != nil {
+		start := time.Now()
+
+		defer func() {
+			duration := time.Since(start)
+			vs.stats.Timing("s3.download_duration", duration, []string{vs.db.name}, 1)
+			vs.stats.Decr("s3.queue_depth", []string{vs.db.name}, 1)
```

Comment:
Oh yes, you're totally right. I thought I would have to take this approach since the encapsulating function is called as a goroutine and I thought that using gauge would introduce the possibility threads writing over each other. As it turns out a lock is taken out and I wasn't tagging metrics so this is actually not an issue.